### ISSUE TITLE
Boolean options don't behave as expected

### ISSUE THREAD

**Issue by Jintram** [woojunshim/TRIAGE#2]
Hi all,

Cool paper!

Just tried the transformation; since I wanted to input a log10(1+exp) transformed matrix, I tried setting `-l False`,

This didn't work. The `log_transform` parameter always remains `True` in the script.

I think this is because the parser behaves weirdly around booleans. Even for the newer version argparse, [they say](https://docs.python.org/3/library/argparse.html#module-argparse) bool shouldn't be used as type converter since "all it does is convert empty strings to False and non-empty strings to True. This is usually not what is desired". This is also what I experienced, the option is now always `True`, whatever the input.

I think you might either want to work around this using `choices`, or make it an int which you convert to boolean (users should supply 0 or 1 as value for False/True respectively).

I now updated the script as follows, seems to work:

```
# from optparse import OptionParser
import argparse
```

```
    # Command line options   
    #parser = OptionParser()
    parser = argparse.ArgumentParser(description = 'TRIAGE')
    parser.add_argument('--input_file', '-i', help='input filename')    
    parser.add_argument('--output_file', '-o',help='output filename')
    parser.add_argument('--pseudo', '-p', help='pseudo count', default=1)
    parser.add_argument('--log_transform', '-l', help='log-transformation (set to 1 or 0 for yes/no)', default=1, type=int)
    parser.add_argument('--ref_file', '-r', help='repressive tendency score (RTS) filename', default='human_rts.txt')
    parser.add_argument('--priority_focus', '-f', help='filtering: whether to output ONLY priority genes (set to 1 or 0 for yes/no)', default=0, type=int)
    parser.add_argument('--stats', '-s', help='Number of top genes for the statistical output', default=0)
    parser.add_argument('--bin_size', '-b', help='Bin size by the number of genes', default=100)
    parser.add_argument('--no_permutation', '-u', help='Number of permutations', default=10000)    

    # options = parser.parse_args()[0]
    options = parser.parse_args()
```

Cheers.

**Comment by woojunshim** [comment]
Hi Martijn, 

Thank you for contacting us. Your comments are very useful and we acknowledge issues related to parameter parsing and input file processing. 

To address these issues, I've made changes as follows. 

1. Now parameter values for --log_transform and --priority_focus require an integer input (i.e. 1 for True, 0 for False) as suggested by Martijn.
2. With the input file processing, I've added additional lines of code to make sure every line of the input has the same number of elements. This ensures that the data are correctly imported. Now the script can import the standard R output file (i.e. where the first line has one less elements) as well as the other format.  

Thank you for bringing these up, Martijn. 

Best Regards,
Woo Jun (Chris)
